### PR TITLE
Add `GET /api/core/v3/committee/` route

### LINKED ISSUE #1052: Add `/api/core/v3/committee` route
https://github.com/iotaledger/tips-draft/blob/43b25b17b564bc42963d9f32645a98f1bc2a58c3/tips/TIPS-API/openapi3-core.yaml#L209C3-L209C25

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Adds the get committee route and associated response.

## Links to any relevant issues

Closes #1052

### CODE REVIEW

**Comment by thibault-martinez** [comment]
Any reason this is still a draft?

**Comment by DaughterOfMars** [comment]
Base branch

File: sdk/src/client/node_api/core/routes.rs [hunk 1]
```diff
@@ -401,6 +404,20 @@ impl ClientInner {
     // // RouteControlSnapshotsCreate is the control route to manually create a snapshot files.
     // // POST creates a snapshot (full, delta or both).
     // RouteControlSnapshotsCreate = "/control/snapshots/create"
+
+    // Accounts routes
+
+    /// Checks if the account is ready to issue a block.
+    /// GET /api/core/v3/accounts/{accountId}/congestion
+    pub async fn get_account_congestion(&self, account_id: &AccountId) -> Result<CongestionResponse> {
```
Comment (thibault-martinez) [inline sdk/src/client/node_api/core/routes.rs]: Keep the TIP order please

File: sdk/src/client/node_api/core/routes.rs [hunk 2]
```diff
@@ -91,15 +91,18 @@ impl ClientInner {
             .await
     }
 
-    /// Checks if the account is ready to issue a block.
-    /// GET /api/core/v3/accounts/{accountId}/congestion
-    pub async fn get_account_congestion(&self, account_id: &AccountId) -> Result<CongestionResponse> {
-        let path = &format!("api/core/v3/accounts/{account_id}/congestion");
+    /// Return the information of committee members at the given epoch index. If epoch index is not provided, the
```
Comment (thibault-martinez) [inline sdk/src/client/node_api/core/routes.rs]:
```suggestion
    /// Returns the information of committee members at the given epoch index. If epoch index is not provided, the
```

File: sdk/src/client/node_api/core/routes.rs [hunk 3]
```diff
@@ -91,15 +91,18 @@ impl ClientInner {
...
+    /// current committee members are returned.
+    /// GET /api/core/v3/committee/?epochIndex
```
Comment (thibault-martinez) [inline sdk/src/client/node_api/core/routes.rs]:
```suggestion
    /// GET /api/core/v3/committee?epochIndex
```

File: sdk/src/types/api/core/response.rs [hunk 4]
```diff
@@ -185,6 +185,41 @@ pub struct BaseTokenResponse {
     pub use_metric_prefix: bool,
 }
 
+/// Response of
+/// - GET /api/core/v3/committee
```
Comment (thibault-martinez) [inline sdk/src/types/api/core/response.rs]: We should remove these lists if there is only one element

File: sdk/src/types/api/core/response.rs [hunk 5]
```diff
@@ -185,6 +185,41 @@ pub struct BaseTokenResponse {
...
+/// The validator information of the committee.
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "camelCase")
+)]
+pub struct CommitteeResponse {
+    pub epoch_index: EpochIndex,
```
Comment (thibault-martinez) [inline sdk/src/types/api/core/response.rs:198]: Documentation from the TIP ?

File: sdk/src/client/node_api/core/routes.rs (hunk 3 above)
Comment (DaughterOfMars) [reply sdk/src/client/node_api/core/routes.rs]: This is not how urls are typically formatted

Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs]: Is it not? Something like https://api.testnet.shimmer.network/api/indexer/v1/outputs/basic?address=rms1qqh8nrvdrg08r2gzezmuaup35eygu7pzl7ekw6zx5vkc8j8gxwwnkvz8npl doesn't work anymore with an ending /

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/src/client/node_api/core/routes.rs (hunk 3 above)
Comment (DaughterOfMars) [reply sdk/src/client/node_api/core/routes.rs]: Weird, it should not matter at all.

Comment (DaughterOfMars) [reply sdk/src/client/node_api/core/routes.rs]: The answer [here](https://stackoverflow.com/questions/1617058/ok-to-skip-slash-before-query-string) provides a pretty in-depth explanation, but the long and short is that it should not matter and I find it strange that the service only accepts one and not the other.

Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs]: Thanks. Yeah honestly I don't know why it's the case, but if I had to choose a flavour, I'd choose the one that works with our own stack, without the ending /. I'm resolving this anyway, feel free to address or not.

File: sdk/src/client/node_api/core/routes.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs]: Just making sure you've seen this one

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by abdulmth — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**